### PR TITLE
Integrate Google OAuth2 Login with UMich Email

### PR DESCRIPTION
**Pull request by eashwar**
# Description

The point of this PR is to integrate Google's OAuth2 login to the WolverineRank app. The login system has been designed such that if you do not login with a G-Suite "umich.edu" email, you will be requested to logout and use one.

# Important Additions

This adds two React components, the `LoginButton` and the `LoginPage`. The `LoginPage` is located at the route `/login` and is the place that Google auth redirects after authentication. It has been added to the `routes.jsx` file accordingly.

The `LoginButton` is a component planned to be part of the global navbar, which not only sports an official "Login with Google" button when logged in, but turns into the "Logout" button when the user is logged in. This is not added to the App anywhere yet, as it is intended to become part of the `Navbar` component which will be added in the future. The `LoginButton` also will insert the user's profile info into the Redux store automatically.

# Notes

**:warning: You will need to ask @eashwar for the Google Client ID and Secret for your `.env` file in order for this to work :warning:**

This PR adds one `npm` library, `google-auth-library`. Please be sure that you run `yarn` or some equivalent command to make sure you have the library installed.

This PR adds the "Roboto" font for the purposes of styling the "Logout" button to match the premade Google Login button.

### CODE REVIEW

**Comment by eashwar** [comment]
force push as a result of rebasing onto new dev (which had the addition of classpage as well as the security fixes)

**Comment by eashwar** [comment]
another fpush because i left tokens in the login page and had to rewrite history to hide,

File: src/reducers/login_reducers.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { combineReducers } from 'redux';
+import { loginTypes } from './../constants/action_types';
+
+function setLoginTokens(state = {}, action) {
+    switch (action.type) {
+        case loginTypes.SET_LOGIN_TOKENS: 
+            return {
+                ...state,
+                error: null,
```
Comment (christopherhafke) [inline src/reducers/login_reducers.js:9]: Consider removing

File: src/reducers/login_reducers.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            return state
+    }
+}
+
+function profileInfo(state = {}, action){
+    switch (action.type) {
+        case loginTypes.SET_PROFILE_INFO:
+            return {
+                ...state,
+                error: null,
```
Comment (christopherhafke) [inline src/reducers/login_reducers.js:28]: ""

File: src/reducers/login_reducers.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                profile: action.profile
+            }
+        case loginTypes.CLEAR_PROFILE_INFO:
+            return {
+                ...state,
+                error: null,
```
Comment (christopherhafke) [inline src/reducers/login_reducers.js:34]: ""

File: src/containers/pages/login_page.jsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                    domain: this.props.profile.hd,
+                })
+            })
+            if (this.state.domain === "umich.edu"){
+                window.location = window.location.origin;
+            } else {
+                window.location = window.location.origin + "/login";
+            }
+        }
+        ).catch((err) => console.log(err));
```
Comment (christopherhafke) [inline src/containers/pages/login_page.jsx]: Dispatch error action

File: src/global_components/login_button.jsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            if (refresh_token) {
+                oauth2Client.setCredentials({
+                    refresh_token
+                });
+            } else {
+                this.getLoginUrl(oauth2Client);
+            }
+        }
+        
+        this.setState({oauth2Client});
```
Comment (christopherhafke) [inline src/global_components/login_button.jsx]: Delete this nephew

File: src/containers/pages/login_page.jsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            const getProfileInfoUrl = 'https://www.googleapis.com/oauth2/v2/userinfo';
+            oauth2Client.request({url: getProfileInfoUrl}).then((res) =>
+            {
+                this.props.setProfileInfo(res.data);
+                this.setState({
+                    domain: this.props.profile.hd,
+                })
+            })
+            if (this.state.domain === "umich.edu"){
+                window.location = window.location.origin;
```
Comment (christopherhafke) [inline src/containers/pages/login_page.jsx]: Use session storage to store url of where user came from

File: src/global_components/login_button.jsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+        this.getProfileInfo(oauth2Client);
+    }
+
+    logoutUser() {
+        localStorage.removeItem("wolverinerank-refresh-token");
+        this.props.clearProfileInfo();
+        window.location = window.location.origin;
```
Comment (christopherhafke) [inline src/global_components/login_button.jsx]: Store url of where user came from in session storage

**Review by christopherhafke — CHANGES_REQUESTED** [review]
Mostly redirects

File: src/containers/pages/login_page.jsx [hunk 8]
```diff
@@ -18,6 +18,10 @@ class LoginPage extends React.Component {
         
         const urlParams = new URLSearchParams(this.props.location.search);
         const code = urlParams.get('code');
+        if (code === null){
+            console.log("boop");
```
Comment (christopherhafke) [inline src/containers/pages/login_page.jsx]: Delete this line

**Review by christopherhafke — APPROVED** [review]
Looks sweet

**Pull request merged**